### ISSUE TITLE
Disable zap by default

### ISSUE THREAD

**Issue by timonwong** [timonwong/loggercheck#27]
By inspecting the zap implementation: 

https://github.com/uber-go/zap/blob/0d6a75bccff91a3106f193c9e4e1678738a63f9d/sugar.go#L339-L342

We can see in zap sugar logger, the keys and values can work without pairs (using "Fields" instead)

So in order to reduce the opportunity of false positives, we should disable it by default.

@ttys3 What do you think?

**Comment by ttys3** [comment]
should not be false positives

they have diffrent package

It is zap core.  what we are checking is zap sugar.

zap core have `Field` which wrap `key-value` pair, so it dooes not need this plugin.

I do not think we have to disable that unless it has been confirmed to be a problem.

**Comment by ttys3** [comment]
"in zap sugar logger, the keys and values can work without pairs"
it is true, but this will trigger an error log in zap itself.

And of course this is where the `loggercheck` linter is useful for zap.

Do not need to wait in production logging, then found the not paired key-value fields problem.

**Comment by ttys3** [comment]
Oops

I re-checked the code your point out.

OK. now I got your point.

the code you point out is a special case.

someone use a zap sugared logger and then write zap core `Field` in it.

like this:

```go
logger, _ := zap.NewProduction()
defer logger.Sync() // flushes buffer, if any

sugar := logger.Sugar()
sugar.Infow("failed to fetch URL",
  zap.String("url", url),
  "attempt", 3,
  "backoff", time.Second,
)
```

OK, in current linter implemention, this should got false positives

**Comment by ttys3** [comment]
but I think use a zap `Field` in a zap sugared logger, this coding style is bad.

if the linter point that out, that should be a good thing ?

one should use zap core logger, it they like to write `Field`, and it has better performance

**Comment by timonwong** [comment]
@ttys3 ok, for most use cases our check is just ok. However I'll try to implement a special checker for zap instead, when my hands free
